src/recipe/recipe.service.ts

```diff
@@ -30,7 +36,21 @@ export class RecipeService {
   }
 
   async findRecipeById(recipeId: number): Promise<RecipeResponse> {
-    return this.queryBus.execute(new FindRecipeByIdQuery(recipeId));
+    const cachedData = await this.cacheService.get<RecipeResponse>(
```

Comment:
On the endpoint you are using  `@UseInterceptors(CacheInterceptor)`
If you are using interceptor there is no reason to do this custom implementation.